selfdrive/car/fw_versions.py

```diff
@@ -284,16 +285,24 @@ def match_fw_to_car_exact(fw_versions_dict):
 
 
 def match_fw_to_car(fw_versions, allow_fuzzy=True):
-  fw_versions_dict = build_fw_dict(fw_versions)
-  matches = match_fw_to_car_exact(fw_versions_dict)
-
+  matches = set()
   exact_match = True
-  if allow_fuzzy and len(matches) == 0:
-    matches = match_fw_to_car_fuzzy(fw_versions_dict)
+  versions = get_interface_attr('FW_VERSIONS', ignore_none=True)
+
+  # For each brand, attempt to fingerprint using FW returned from its queries
+  for brand in versions.keys():
+    fw_versions_dict = build_fw_dict(fw_versions, filter_brand=brand)
+    matches = match_fw_to_car_exact(fw_versions_dict)
+
+    if allow_fuzzy and len(matches) == 0:
+      matches = match_fw_to_car_fuzzy(fw_versions_dict)
+
+      # Fuzzy match found
+      if len(matches) == 1:
```

Comment:
hmm, this doesn't look quite right. should only be able to fuzzy fingerprint after exhausting all brands for exact matches

Reply:
Fixed